File: JPR/src/controllers/analizador/Instrucciones/Imprimir.ts
import { Instruccion } from "../Abstract/Instruccion";
import Excepcion from "../Excepciones/Excepcion";
import Arbol from "../tablaSimbolos/Arbol";
import tablaSimbolos from "../tablaSimbolos/tablaSimbolos";
import Tipo, { tipos } from "../tablaSimbolos/Tipo";


export default class Imprimir extends Instruccion{
    private expresion: Instruccion;

    constructor(expresion:Instruccion, linea:Number, columna:Number){
        super(new Tipo(tipos.CADENA),linea, columna);
        this.expresion = expresion;
    }

    public interpretar(tree:Arbol, table:tablaSimbolos){
        var value = this.expresion.interpretar(tree, table); //OBTIENE EL VALOR

        if(value instanceof Excepcion) return value;

        tree.updateConsola(value+"");
    }
}

File: JPRJava/JPR/src/Instrucciones/Imprimir.java
package Instrucciones;

import Abstract.AST;
import Abstract.nodoAST;
import Excepciones.Excepcion;
import TablaSimbolos.Arbol;
import TablaSimbolos.Tabla;

/**
 *
 * @author josef
 */
public class Imprimir extends AST {

    private final AST expresion;

    public Imprimir(AST expresion, int fila, int columna) {
        this.expresion = expresion;
        this.fila = fila;
        this.columna = columna;
    }

    @Override
    public Object interpretar(Tabla tabla, Arbol tree) {
        Object value = expresion.interpretar(tabla, tree);
        if (value instanceof Excepcion) return value;
        
        tree.updateConsola(value + "");
        
        return null;
    }
    
    @Override
    public nodoAST getNodo() {
        nodoAST nodo  = new nodoAST("PRINT");
        nodo.agregarHijo("print");
        nodo.agregarHijo("(");
        nodo.agregarHijo(expresion.getNodo());
        nodo.agregarHijo(")");
        return nodo;
    }
}
